src/messages/mod.rs

```diff
@@ -318,3 +316,126 @@ impl GroupSecrets {
         Ok(buffer.to_vec())
     }
 }
+
+/// PublicGroupState
+///
+/// ```text
+/// struct {
+///     CipherSuite cipher_suite;
+///     opaque group_id<0..255>;
+///     uint64 epoch;
+///     opaque tree_hash<0..255>;
+///     opaque interim_transcript_hash<0..255>;
+///     Extension extensions<0..2^32-1>;
+///     HPKEPublicKey external_pub;
+///     uint32 signer_index;
+///     opaque signature<0..2^16-1>;
+/// } PublicGroupState;
+/// ```
+#[derive(PartialEq, Debug)]
+pub struct PublicGroupState {
+    pub ciphersuite: CiphersuiteName,
+    pub group_id: GroupId,
+    pub epoch: GroupEpoch,
+    pub tree_hash: Vec<u8>,
+    pub interim_transcript_hash: Vec<u8>,
+    pub extensions: Vec<Box<dyn Extension>>,
+    pub external_pub: HPKEPublicKey,
+    pub signer_index: LeafIndex,
+    pub signature: Signature,
+}
+
+impl PublicGroupState {
+    /// Creates a new `PublicGroupState` struct from the current internal state
+    /// of the group and signs it.
+    pub(crate) fn new(
+        mls_group: &MlsGroup,
+        credential_bundle: &CredentialBundle,
+    ) -> Result<Self, CredentialError> {
+        let ciphersuite = mls_group.ciphersuite();
+        let (_external_priv, external_pub) = mls_group
```

Comment:
This should go into the key schedule but I'll do this in my PR that reworks it.

Reply:
I thought about that as well, but decided to expose it as a function of `MlsGroup`. The reason is that it does a keypair derivation, which is a fairly expensive operation and it's only needed when External Commits are supported and not otherwise.
Ideally we could have this as part of the configuration (see #206).